### PR TITLE
Convert to python3

### PR DESCRIPTION
**Pull request by dbnicholson**
This has a python3 conversion with some other cleanups and a testsuite to give a little bit of assurance that it works.

### CODE REVIEW

**Comment by dbnicholson** [comment]
This is working and ready for review. There are really 2 parts, though. The actual python3 port ends at `TarImporter: Use TemporaryDirectory for automatic cleanup`. The rest is some fixes and improvements to ultimately allow testing. In order to feel confident that the python3 port was working, I really wanted actual tests instead of manual tests. However, I have confirmed with the manual test scripts that the changes work and I'm happy to split up the PR if someone doesn't want to take on the whole thing.

There is actually one breaking change in `Merge flatpak-import.conf into ostree-upload-server.conf`. Previously there were 2 configuration files with `flatpak-import.conf` providing the settings for actually importing the flatpak commits. That was originally done because the importer was a standalone tool that later got integrated as a subpackage. Testing the server was difficult since loading of `flatpak-import.conf` happened within the server without a method to specify where it was loaded from. I think that could probably have been fixed, but instead I opted to fold the settings into the main configuration file and allow providing a path to the main config file.

File: ostree_upload_server/bundle_importer.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def import_bundle(bundle, repository, gpg_homedir=None, keyring=None,
         sign_key = sign_key or defaults.get('sign_key', None)
 
         logging.info("Starting the bundle import process...")
-        for arg in inspect.getargspec(BundleImporter.import_bundle)[0]:
+        for arg in inspect.getfullargspec(BundleImporter.import_bundle)[0]:
```
Comment (wjt) [inline ostree_upload_server/bundle_importer.py:18]: This is a pretty weird way to log what is rather a short list of parameters.

**Review by wjt — APPROVED** [review]
Magnificent. I guess the breaking change is manageable as a one-off when deploying the new server.

File: ostree_upload_server/bundle_importer.py (hunk 1 above)
Comment (dbnicholson) [reply ostree_upload_server/bundle_importer.py:18]: Indeed. I considered tearing it out, but opted to leave it working the way it is.

**Pull request merged**